### PR TITLE
[WIP] DO NOT MERGE - allows for variables to be managed by adrenaline as a function of props

### PR DESCRIPTION
**Pull request by pcwinters**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -265,12 +265,13 @@ export default createDumbComponent(TodoList, {
...
 ```
 
-### `createSmartComponent(Component, { initialVariables, query, mutations })`
+### `createSmartComponent(Component, { initialVariables, variables, query, mutations })`
 
 This function is the main building block for your application. It is similar to [react-redux smart component](https://github.com/rackt/react-redux#smart-components-are-connect-ed-to-redux) but with ability to declare your data query with GraphQL.
 
   - `Component`: Its your component which would be wrapped.
-  - `initialVariables`: This is an are your arguments which would be applied to your query. You can declare it as a plain object or as a function of props.
+  - `initialVariables`: Optional. This is an are your arguments which would be applied to your query. You can declare it as a plain object or as a function of props. When variables have changed, your component will need to notify adrenaline by invoking this.setVariables(variables).
```
Comment (markmsmith) [inline README.md:273]: Typo: "This is an are your"

Comment (pcwinters) [inline README.md:273]: note that it wasn't **my** typo, but pre-existing. I'll fix.

File: src/components/createSmartComponent.jsx [hunk 2]
```diff
@@ -59,10 +61,23 @@ export default function createSmartComponent(DecoratedComponent, specs) {
...
+        if(isFunction(specs.variables)
+            && nextProps != this.props
+            && !shallowEqual(nextProps, this.props)){
+            this.setVariables(specs.variables(nextProps))
+        }
+    }
+
     /*shouldComponentUpdate(nextProps) {
       return !shadowEqualScalar(this.props, nextProps);
     }*/
```
Comment (markmsmith) [inline src/components/createSmartComponent.jsx:83]: Old code?

Comment (pcwinters) [inline src/components/createSmartComponent.jsx:83]: lol look at the diff, man

**Comment by markmsmith** [comment]
Minor comments, otherwise LGTM.

**Comment by pdillon** [comment]
LGTM

**Pull request closed without merging**